### PR TITLE
Show ideas

### PR DESCRIPTION
**Pull request by MiBBy-4**

### CODE REVIEW

File: src/Idea/IdeaForm.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { TextField, Button } from '@mui/material';
+import React, { Component } from 'react';
+
+class IdeaForm extends Component {
```
Comment (AndreyAnischik) [inline src/Idea/IdeaForm.jsx:4]: use functional components and hooks

File: src/Idea/IdeaList.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react';
+import IdeaItem from './IdeaItem';
+import IdeaForm from './IdeaForm';
+
+const apiURL = 'http://localhost:3000/api/v1/ideas';
+
+class IdeaList extends Component {
```
Comment (AndreyAnischik) [inline src/Idea/IdeaList.jsx:7]: functional component

**Review by AndreyAnischik — CHANGES_REQUESTED** [review]

File: src/Idea/IdeaForm.jsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  const [apiURL, setApiURL] = useState([props.apiURL]);
+  const [name, setName] = useState('');
+  const [description, setDescription] = useState('');
+  const [problem, setProblem] = useState('');
+  const [sphere, setSphere] = useState('');
+  const [geo_focus, setGeoFocus] = useState('');
+  const [investor_requirements, setInvestorRequirements] = useState('');
+
+  function handleIdeaChange(event) {
+    setName(event.target.value);
```
Comment (AndreyAnischik) [inline src/Idea/IdeaForm.jsx:14]: make object destructuring (https://dmitripavlutin.com/javascript-object-destructuring/) before setting the values and move it to a separate var

Comment (AndreyAnischik) [inline src/Idea/IdeaForm.jsx:14]: won't it set same values for all the fields?

File: src/Idea/IdeaForm.jsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    setDescription(event.target.value);
+    setProblem(event.target.value);
+    setSphere(event.target.value);
+    setGeoFocus(event.target.value);
+    setInvestorRequirements(event.target.value);
+  }
+
+  async function formSubmit(formData) {
+    const data = new FormData(formData);
+    await fetch(apiURL, {
```
Comment (AndreyAnischik) [inline src/Idea/IdeaForm.jsx:24]: make single entry for all endpoints (move fetch functionality to a separate file and use it from other components too

File: src/Idea/IdeaList.jsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+const apiURL = 'http://localhost:3000/api/v1/ideas';
+
+function IdeaList() {
+  const [items, setItems] = useState([]);
+
+  function getIdeas() {
+    fetch(apiURL)
+      .then((response) => response.json())
+      .then((responseItems) => {
```
Comment (AndreyAnischik) [inline src/Idea/IdeaList.jsx:13]: move it a separate file (the same file as for POST requests)

File: src/Idea/IdeaList.jsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from 'react';
+import IdeaItem from './IdeaItem';
+import IdeaForm from './IdeaForm';
+
+const apiURL = 'http://localhost:3000/api/v1/ideas';
```
Comment (AndreyAnischik) [inline src/Idea/IdeaList.jsx:5]: move BASE_URL to an .env file

File: src/Idea/IdeaList.jsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    let updatedItems = items;
+    updatedItems.unshift(item);
+    setItems({
+      items: updatedItems,
+    });
+  }
+
+  return (
+    <div>
+      <IdeaForm apiURL={apiURL} updateIdeaList={updateIdeaList} />
```
Comment (AndreyAnischik) [inline src/Idea/IdeaList.jsx:32]: you don't need to pass apiUrl when it will be in a constant and a separate file

**Review by AndreyAnischik — CHANGES_REQUESTED** [review]

File: src/Idea/IdeaForm.jsx (hunk 3 above)
Comment (MiBBy-4) [reply src/Idea/IdeaForm.jsx:14]: Yeap, I think it. I did it only for test, I will change it.

File: src/Idea/IdeaFetch.jsx [hunk 8]
```diff
@@ -0,0 +1,16 @@
+export async function postFetchResponse(data) {
+  console.log('Inside postFetchResponse');
```
Comment (AndreyAnischik) [inline src/Idea/IdeaFetch.jsx:2]: remove log please

File: src/Idea/IdeaForm.jsx [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  const [name, setName] = useState('');
+  const [description, setDescription] = useState('');
+  const [problem, setProblem] = useState('');
+  const [sphere, setSphere] = useState('');
+  const [geo_focus, setGeoFocus] = useState('');
+  const [investor_requirements, setInvestorRequirements] = useState('');
+
+  function handleIdeaChange(event) {
+    const { target: { value } } = event;
+    setName(event);
```
Comment (AndreyAnischik) [inline src/Idea/IdeaForm.jsx:16]: you need to set value here

File: src/Idea/IdeaForm.jsx [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    setDescription(event);
+    setProblem(event);
+    setSphere(event);
+    setGeoFocus(event);
+    setInvestorRequirements(event);
+  }
+
+  async function formSubmit(formData) {
+    const data = new FormData(formData);
+    console.log('Inside formSubmit');
```
Comment (AndreyAnischik) [inline src/Idea/IdeaForm.jsx:26]: remove log

File: src/Idea/IdeaForm.jsx [hunk 11]
```diff
@@ -0,0 +1,50 @@
+import { TextField, Button } from '@mui/material';
+import { React, useState } from 'react';
+import { postFetchResponse } from './IdeaFetch';
+
+function IdeaForm(props) {
+  const [apiURL, setApiURL] = useState(process.env.REACT_APP_API_URL);
```
Comment (AndreyAnischik) [inline src/Idea/IdeaForm.jsx:6]: do you use this var?

**Review by AndreyAnischik — CHANGES_REQUESTED** [review]

File: src/Idea/IdeaForm.jsx (hunk 11 above)
Comment (MiBBy-4) [reply src/Idea/IdeaForm.jsx:6]: No, sorry. I always forget about this

**Review by AndreyAnischik — CHANGES_REQUESTED** [review]

**Pull request merged**